lib/armadietto.js

```diff
@@ -26,14 +26,20 @@ class Armadietto {
     this._allow = options.allow || {};
     this._cacheViews = options.cacheViews !== false;
     this._basePath = options.basePath || '';
-    this._httpServer = null;
-    this._httpsServer = null;
+    this._server = null;
     this.init();
   }
 
   async init () {
+    let _http = null;
+    let _https = null;
+    let resolveFn;
+    this._server = new Promise((resolve) => {
+      resolveFn = resolve;
```

Comment:
Why not move the rest of the `init` body inside of this promise? I think it would be more clear as to what the promise is waiting on, rather than looking for `resolveFn`